pkg/http/core/kubernetes/deploy.go

```diff
@@ -99,30 +109,63 @@ func Deploy(c *gin.Context, dm DeployManifestRequest) {
 		// If the kind is a job, its name is not set, and generateName is set,
 		// generate a name for the job as `apply` will throw the error
 		// `resource name may not be empty`.
+		name := u.GetName()
+
 		if strings.EqualFold(u.GetKind(), "job") {
-			name := u.GetName()
 			generateName := u.GetGenerateName()
 
 			if name == "" && generateName != "" {
 				u.SetName(generateName + rand.String(randNameNumber))
 			}
 		}
 
-		name := u.GetName()
-
-		err = kc.AddSpinnakerAnnotations(u, dm.Moniker.App)
+		err = kc.AddSpinnakerAnnotations(u, application)
 		if err != nil {
 			clouddriver.Error(c, http.StatusInternalServerError, err)
 			return
 		}
 
-		err = kc.AddSpinnakerLabels(u, dm.Moniker.App)
+		err = kc.AddSpinnakerLabels(u, application)
 		if err != nil {
 			clouddriver.Error(c, http.StatusInternalServerError, err)
 			return
 		}
 
+		if kc.IsVersioned(u) {
+			kind := strings.ToLower(u.GetKind())
+			namespace := u.GetNamespace()
+			lo := metav1.ListOptions{
+				LabelSelector:  kubernetes.LabelKubernetesName + "=" + application,
```

Comment:
Add an additional label selector here to only select resources that are Spinnaker versioned. This will greatly help with filter efficiency.

See comment [here](https://github.com/homedepot/go-clouddriver/pull/3#issuecomment-742037259).